src/components/toniq-checkbox/toniq-checkbox.component.ts

```diff
@@ -0,0 +1,97 @@
+import {css, defineElementEvent, defineFunctionalElement, html} from 'element-vir';
+
+export const ToniqCheckbox = defineFunctionalElement({
+    tagName: 'toniq-checkbox',
+    props: {
+        text: '',
+        checked: false,
+    },
+    events: {
+        change: defineElementEvent<boolean>(),
+    },
+    styles: css`
+        .toniq-checkbox {
+            display: flex;
+            align-items: center;
+            cursor: pointer;
+        }
+
+        .toniq-checkbox:hover input[type='checkbox']:not(:checked) + .checkbox,
+        .toniq-checkbox:hover input[checked='false'] + .checkbox {
+            background-color: var(--toniq-checkbox-background-color-hover, #e0dfdb);
+        }
+
+        .toniq-checkbox:hover input[type='checkbox']:checked + .checkbox,
+        .toniq-checkbox:hover input[checked='true'] + .checkbox {
+            background-color: var(--toniq-checkbox-background-color-checked-hover, #02c58c);
+        }
+
+        .toniq-checkbox:hover input[type='checkbox']:not(:checked) + .checkbox + .label,
+        .toniq-checkbox:hover input[checked='false'] + .checkbox + .label {
+            color: var(--toniq-checkbox-label-color-hover, #303030);
+        }
+
+        .toniq-checkbox:hover input[type='checkbox']:checked + .checkbox + .label,
+        .toniq-checkbox:hover input[checked='true'] + .checkbox + .label {
+            color: var(--toniq-checkbox-label-color-checked-hover, #02c58c);
+        }
+
+        .label {
+            font-family: var(--toniq-font, 'Rubik');
+            font-style: normal;
+            font-weight: 700;
+            font-size: 16px;
+            line-height: 24px;
+            color: var(--toniq-secondary-interaction-text-color, black);
+            margin-left: 16px;
+
+            -webkit-touch-callout: none; /* iOS Safari */
+            -webkit-user-select: none; /* Safari */
+            -khtml-user-select: none; /* Konqueror HTML */
+            -moz-user-select: none; /* Old versions of Firefox */
+            -ms-user-select: none; /* Internet Explorer/Edge */
+            user-select: none; /* Non-prefixed version, currently supported by Chrome, Edge, Opera and Firefox */
+        }
+
+        input[type='checkbox'] {
+            display: none;
+        }
+
+        input[type='checkbox']:checked + span,
```

Comment:
Please use classes here to indicate that this is intended for the `.checkbox` span:
```suggestion
        input[type='checkbox']:checked + .checkbox,
```